types/react-router/index.d.ts

```diff
@@ -39,7 +39,7 @@ declare module 'react-router' {
 
 
   interface PromptProps {
-    message: string | ((location: H.Location) => void);
+    message: string | ((location: H.LocationDescriptor) => void);
     when?: boolean;
```

Comment:
This change is probably also not correct because according to the [documentation](https://reacttraining.com/react-router/core/api/Prompt/message-func), if `message` is a function, it 

> Will be called with the next location and action the user is attempting to navigate to. Return a string to show a prompt to the user or true to allow the transition.

Because of the library's internal consistency, most likely the `location` passed to `message` will be a complete `H.Location` object instead of `H.LocationDescriptor`.